### PR TITLE
fix(gpu): fix lut size in are_all_comparison_blocks_true

### PR DESCRIPTION
**Pull request by agnesLeroy**
closes: _please link all relevant issues_

### Check-list:

* [ ] Tests for the changes have been added (for bug fixes / features)
* [ ] Docs have been added / updated (for bug fixes / features)
* [ ] Relevant issues are marked as resolved/closed, related issues are linked in the description
* [ ] Check for breaking changes (including serialization changes) and add them to commit message following the conventional commit [specification][conventional-breaking]

[conventional-breaking]: https://www.conventionalcommits.org/en/v1.0.0/#commit-message-with-description-and-breaking-change-footer

### CODE REVIEW

**Comment by agnesLeroy** [comment]
This one fixes the out of memory error Antoniu spotted in the erc20 throughput benchmarks.

File: backends/tfhe-cuda-backend/cuda/include/integer/integer_utilities.h [hunk 1]
```diff
@@ -3079,7 +3079,7 @@ template <typename Torus> struct int_are_all_block_true_buffer {
                                            streams[0], gpu_indexes[0]);
       is_max_value =
           new int_radix_lut<Torus>(streams, gpu_indexes, gpu_count, params, 2,
-                                   num_radix_blocks, allocate_gpu_memory);
+                                   max_value, allocate_gpu_memory);
```
Comment (guillermo-oyarzun) [inline backends/tfhe-cuda-backend/cuda/include/integer/integer_utilities.h:3082]: shouldn't this be num_chunks instead?

Comment (agnesLeroy) [reply backends/tfhe-cuda-backend/cuda/include/integer/integer_utilities.h:3082]: Oh  you're right it should be `max_chunks` here

Comment (agnesLeroy) [reply backends/tfhe-cuda-backend/cuda/include/integer/integer_utilities.h:3082]: We apply the lut with this:
```
      integer_radix_apply_univariate_lookup_table_kb<Torus>(
          streams, gpu_indexes, gpu_count, tmp_out, accumulator, bsks, ksks,
          num_chunks, lut);
```
so num_chunks is the amount of inputs, and we have to allocate with the maximum possible num_chunks vaue, which is `max_chunks`

**Review by guillermo-oyarzun — APPROVED** [review]

**Pull request merged**